editor/components/autosave-monitor/index.js

```diff
@@ -6,10 +6,17 @@ import { withSelect, withDispatch } from '@wordpress/data';
 
 export class AutosaveMonitor extends Component {
 	componentDidUpdate( prevProps ) {
-		const { isDirty, isSaveable } = this.props;
-		if ( prevProps.isDirty !== isDirty ||
-				prevProps.isSaveable !== isSaveable ) {
-			this.toggleTimer( isDirty && isSaveable );
+		const { isDirty, isAutosaveable, isAutosaving } = this.props;
+
+		// Prevent autosaves if an autosave is in progress.
+		if ( isAutosaving ) {
+			return;
```

Comment:
`return`ing here is not preventing, since it's the behavior of `toggleTimer` to either schedule or cancel a scheduled autosave. So if a `toggleTimer( true )` had been called previously and on the next `componentDidUpdate` `isAutosave` is `true`, the scheduled autosave won't be stopped.

Should we want this to be considered instead as part of the condition below? i.e.

```js
if (
	prevProps.isDirty !== isDirty ||
	prevProps.isAutosaveable !== isAutosaveable ||
	prevProps.isAutosaving !== isAutosaving
) {
	this.toggleTimer( isDirty && isAutosaveable && ! isAutosaving );
}
```

Reply:
that would work, although as you point out this only affects the setting of the timer. to truly prevent a duplicate request, we need to check if an autosaving is happening right before we try to initiate a new one.